src/Bridges/ApplicationLatte/TemplateFactory.php

```diff
@@ -48,7 +51,11 @@ public function __construct(ILatteFactory $latteFactory, Nette\Http\IRequest $ht
 	public function createTemplate(UI\Control $control = NULL)
 	{
 		$latte = $this->latteFactory->create();
-		$template = new Template($latte);
+		$template = new $this->templateClass($latte);
+		if (!$template instanceof UI\ITemplate) {
```

Comment:
This check should be moved to `setTemplateClass` (see is_a with third argument set to TRUE)